### PR TITLE
Create new endPoint for MentorController

### LINKED ISSUE #303: Add GetAllActiveCourses endpoint in CourseController
Add new endtpoint in courses controller. This endpoint must return only active courses.

### PR DESCRIPTION
**Pull request by QuestroL**
Add new Endpoint to mentorController (GetAllLessonForMentor)

### CODE REVIEW

File: CharlieBackend.Business/Services/MentorService.cs [hunk 1]
```diff
@@ -93,6 +97,22 @@ public async Task<IList<MentorDto>> GetAllActiveMentorsAsync()
             return mentors;
         }
 
+        public async Task<Result<IList<LessonDto>>> GetAllLessonsForMentor(long mentorId)
+        {
+            if (mentorId == default)
+            {
+                return Result<IList<LessonDto>>.GetError(ErrorCode.ValidationError, "Write Mentod Id please");
```
Comment (filinskyi) [inline CharlieBackend.Business/Services/MentorService.cs]: In error message fix "Mentod" to "Mentor"

File: CharlieBackend.Api/Controllers/MentorsController.cs [hunk 2]
```diff
@@ -159,5 +160,20 @@ public async Task<ActionResult> DisableMentor(long id)
 
             return disabledMentorModel.ToActionResult();
         }
+
+        /// <summary>
+        /// Returns list of lessons  for mentor
```
Comment (filinskyi) [inline CharlieBackend.Api/Controllers/MentorsController.cs:167]: Remove extra space between "lessons" and "for"

File: CharlieBackend.Api/Controllers/MentorsController.cs [hunk 3]
```diff
@@ -159,5 +160,20 @@ public async Task<ActionResult> DisableMentor(long id)
...
+        /// </summary>
+        /// <param name="id"></param>
+        /// <response code="200">Successful return of lessons list of given student</response>
```
Comment (filinskyi) [inline CharlieBackend.Api/Controllers/MentorsController.cs]: Replace "student" to "mentor"

File: CharlieBackend.Business/Services/MentorService.cs [hunk 4]
```diff
@@ -93,6 +97,22 @@ public async Task<IList<MentorDto>> GetAllActiveMentorsAsync()
...
+            }
+            if (await _unitOfWork.MentorRepository.GetByIdAsync(mentorId) == null)
+            {
+                return Result<IList<LessonDto>>.GetError(ErrorCode.ValidationError, $"Mentod with id {mentorId} is not Found");
```
Comment (filinskyi) [inline CharlieBackend.Business/Services/MentorService.cs]: Do the same fix as in line 104

File: CharlieBackend.Api/Controllers/LessonsController.cs [hunk 5]
```diff
@@ -102,6 +102,7 @@ public async Task<ActionResult<List<StudentLessonDto>>> GetStudentLessons(long i
             return lessons.ToActionResult();
         }
 
+
```
Comment (filinskyi) [inline CharlieBackend.Api/Controllers/LessonsController.cs:105]: Remove this blank line

**Review by filinskyi — CHANGES_REQUESTED** [review]
Small changes required

File: CharlieBackend.Api.UnitTes/MentorServiceTests.cs [hunk 6]
```diff
@@ -56,11 +56,13 @@ public async Task CreateMentorAsync()
...
             _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorRepositoryMock.Object);
 
+            var lessonServiceMock = new Mock<ILessonService>();
+
             var mentorService = new MentorService(
                 _accountServiceMock.Object,
                 _unitOfWorkMock.Object,
                 _mapper,
-                _notificationServiceMock.Object);
+                _notificationServiceMock.Object, lessonServiceMock.Object);
```
Comment (Hlib-mats) [inline CharlieBackend.Api.UnitTes/MentorServiceTests.cs]: lessonServiceMock.Object must be on a next line and change variable name to _lessonServiceMock

**Review by Hlib-mats — CHANGES_REQUESTED** [review]

File: CharlieBackend.Data/Repositories/Impl/LessonRepository.cs [hunk 7]
```diff
@@ -23,6 +23,14 @@ public new Task<List<Lesson>> GetAllAsync()
                     .ToListAsync();
         }
 
+        public async Task<List<Lesson>> GetAllLessonsForMentor(long mentodId)
```
Comment (filinskyi) [inline CharlieBackend.Data/Repositories/Impl/LessonRepository.cs]: fix argument name from "mentodId" to "mentorId"

File: CharlieBackend.Data/Repositories/Impl/Interfaces/ILessonRepository.cs [hunk 8]
```diff
@@ -8,5 +8,7 @@ namespace CharlieBackend.Data.Repositories.Impl.Interfaces
     public interface ILessonRepository : IRepository<Lesson>
     {
         public Task<IList<StudentLessonDto>> GetStudentInfoAsync(long studentId);
+
+        Task<List<Lesson>> GetAllLessonsForMentor(long mentodId);
```
Comment (filinskyi) [inline CharlieBackend.Data/Repositories/Impl/Interfaces/ILessonRepository.cs]: fix argument name from "mentodId" to "mentorId"

**Review by filinskyi — CHANGES_REQUESTED** [review]
small fix needed

**Review by filinskyi — APPROVED** [review]

**Review by Hlib-mats — APPROVED** [review]

**Review by CptHouston — APPROVED** [review]

**Pull request merged**